[Docs] Restructure of Vizro-AI dashboard docs (#883)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/run-vizro-ai-dashboard.md

        from vizro.tables import dash_ag_grid

        ########## Data Imports ##########
        #####!!! UNCOMMENT BELOW !!!######
        # data_manager["restaurant_bills"] = ===> Fill in here <===

        ###### Callable definitions ######

        ########## Object code ###########
        dashboard = vm.Dashboard(
           pages=[
               vm.Page(
                   id="Tips Data Visualization",
                   components=[
                       vm.Card(
                           id="tips_card_tips_data_visualization",
                           type="card",
                           text="This is Tips dataset",
                           href="",
                       ),
                       vm.AgGrid(
                           id="tips_table_tips_data_visualization",
                           figure=dash_ag_grid(data_frame="restaurant_bills"),
                       ),
                   ],
                   title="Tips Data Visualization",
                   layout=vm.Layout(grid=[[0, 1, 1, 1]]),
                   controls=[],
               )
           ],
           title="Tips Dataset Overview",
        )
        ```

> To use the above code, you will still need to add three simple steps.

Reviewer: ```suggestion
To use the above code, you will still need to add three simple steps:
```

> 1. Import your data into the Python environment.

Reviewer: These can be bullets since you don't have to execute them in a particular order (you just need to do them all), right?

> 2. Uncomment data manager instance and map the imported data. See Vizro guide on [connecting dashboard to data](https://vizro.readthedocs.io/en/stable/pages/user-guides/data/).

Reviewer: I'm still not totally sure I understand why, or what code. Would it make sense to show the code you need to uncomment and explain what it's doing (and why it's commented out in the generated version).

3. Add following code snippet at the end of the file `Vizro().build(dashboard).run()` to launch the dashboard.

> Detailed guidance is provided in [dashboard generation tutorial](https://vizro.readthedocs.io/projects/vizro-ai/en/vizro-ai-0.3.2/pages/tutorials/quickstart/)

Reviewer: ```suggestion
Detailed guidance is provided in [dashboard generation tutorial](https://vizro.readthedocs.io/projects/vizro-ai/en/latest/pages/tutorials/quickstart/)
```

Reviewer: Is there a reason for using a specific version of the docs?

# How to run Vizro-AI dashboard

> This guide offers insights into different ways of running `VizroAI.dashboard` to generate Vizro dashboard.

Reviewer 2: ```suggestion
This guide offers insights into different ways of running `VizroAI.dashboard` to generate Vizro dashboards from natural language prompts.
```

??? note "Note: API key"

    Make sure you have followed the [LLM setup guide](../user-guides/install.md#set-up-access-to-a-large-language-model) and thatyour api key is set up in a `.env` file in the same folder as your Notebook file (`.ipynb`).

## Run Vizro-AI dashboard

### 1. Prepare the data and user prompt
```py
import vizro.plotly.express as px

df = px.data.tips()

user_question = """
Create a one-page dashboard layout with the following components:

1. Card:
   - Position: Left of the page
   - Size: Takes up 1/4 of the total page width
   - Content: Display the text "This is Tips dataset"

2. Table:
   - Position: Right of the card
   - Size: Takes up the remaining 3/4 of the page width
   - Content: Display the Tips dataset
"""
```

> ### 2. Generate and launch the dashboard

Reviewer 2: Maybe we could try to avoid to repeat the prompt here? Then it is less text that is rewritten?

I always prefer on comprehensive example that does it all

dashboard = vizro_ai.dashboard([df], user_question)
```
This triggers the dashboard building process. Once Vizro-AI finishes the dashboard generation process, you can now launch the dashboard.

!!! example "Generated dashboard"

    === "Code"
        ```py
        import vizro.plotly.express as px
        from vizro_ai import VizroAI
        from vizro import Vizro

        df = px.data.tips()
        user_question = """
        Create a one-page dashboard layout with the following components:

        1. Card:
           - Position: Left of the page
           - Size: Takes up 1/4 of the total page width
           - Content: Display the text "This is Tips dataset"

        1. Table:
           - Position: Right of the card
           - Size: Takes up the remaining 3/4 of the page width
           - Content: Display the Tips dataset
        """
        vizro_ai = VizroAI(model="gpt-4o-mini")
        dashboard = vizro_ai.dashboard([df], user_question)
        Vizro().build(dashboard).run()
        ```

    === "Result"
        [![VizroAIDashboardPage1]][VizroAIDashboardPage1]

    [VizroAIDashboardPage1]: ../../assets/user_guides/dashboard/dashboard2_page1.png

> ## Retrieve the Python code of the dashboard

Reviewer 2: we should explain that people still need to set the data manager

Reviewer: Yeah, what is that about. I've seen it in the output code from Vizro-AI but haven't understood what to do. Please could you add a paragraph or let me know and I'll do it.

> To illustrate the process, lets re-use the example provided above.

Reviewer: ```suggestion
To illustrate the process, lets use the example above.
```

| **Layout**           | [Layout](https://vizro.readthedocs.io/en/stable/pages/user-guides/layouts/)                                                              | ✔           |
| **Dashboard header** | [Dashboard title](https://vizro.readthedocs.io/en/stable/pages/user-guides/dashboard/)                                                   | ✔           |
|                      | [Logo](https://vizro.readthedocs.io/en/stable/pages/user-guides/dashboard/)                                                              | ✖           |
| **Actions**          | [Pre-defined actions](https://vizro.readthedocs.io/en/stable/pages/user-guides/actions/#pre-defined-actions/)                            | ✖           |
|                      | [Filter interaction between charts](https://vizro.readthedocs.io/en/stable/pages/user-guides/actions/#filter-data-by-clicking-on-chart/) | ✖           |
|                      | [Custom actions](https://vizro.readthedocs.io/en/stable/pages/user-guides/actions/#custom-actions/)                                      | ✖           |

> If a feature you need for your dashboard isn't currently supported by Vizro-AI you can retrieve the dashboard code and add the missing components before running the dashboard.

Reviewer: ```suggestion
If a feature you need for your dashboard isn't currently supported by Vizro-AI you can retrieve the dashboard code and add it by hand before running the dashboard.
```
